File: Python Pattern Programs/Numeric Patterns/Pattern 33.py
size = 4
p = 1
for x in range(size, -(size + 1), -1):
    for y in range(1, abs(x) + 1):
        print(" ", end="")

    if (x >= 0):
        p = 1
    else:
        p = y + 1

    for z in range(size, abs(x) - 1, -1):
        print(str(p) + " ", end="")
        p += 1
    print()

File: C# Pattern Programs/Alphabetic Patterns/Pattern 33.cs
using System;

class Pattern_79
 {
  public static void Main()
   {
    int size = 4;
    int p = 0; // print value

    for (int i = size; i >= -size; i--)
     {
      for (int j = 1; j <= Math.Abs(i); j++)
       {
        Console.Write(" ");
       }
        for (int k = size; k >= Math.Abs(i); k--)
         {
          Console.Write((char)(p + 65) + " ");
         }

        if (i > 0)
         {
          p++;
         }
        else
         {
          p--;
         }

     Console.WriteLine();
     }
     Console.ReadKey(true);
   }
 }
